### PR TITLE
LamBui-Handle Playlist

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/src/main/java/internship/asiantech/a2018summerfinal/adapter/PlaylistChoiceAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        val song = listSongs[position]
+        holder as ListSongsChoiceViewHolder
+        holder.tvNameSongListChoice.text = song.name
+        holder.tvAristListChoice.text = song.author
+        holder.rbSonglistChoice.isChecked = listChoice[position]
+    }
+
+    inner class ListSongsChoiceViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/adapter/PlaylistChoiceAdapter.kt]: ya shouldn't use inner class ,use class instead

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt [hunk 2]
```diff
@@ -1,22 +1,130 @@
...
+import internship.asiantech.a2018summerfinal.database.PlaylistUpdater
+import internship.asiantech.a2018summerfinal.database.model.Playlist
+import internship.asiantech.a2018summerfinal.fragment.dialog.EventAddPlaylistListener
+import internship.asiantech.a2018summerfinal.ui.ListSongInPlaylistActivity
+import kotlinx.android.synthetic.main.fragment_playlist.*
+
+@Suppress("DEPRECATION")
+class PlaylistFragment : Fragment(), EventAddPlaylistListener, EventClickItemOpenListSong {
+
+
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt]: delete  line

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt [hunk 3]
```diff
@@ -1,22 +1,130 @@
...
+    private var listPlaylist: MutableList<Playlist> = mutableListOf()
+    private lateinit var mAdapterPlaylist: PlaylistAdapter
+    private lateinit var mRecyclerviewPlaylist: RecyclerView
+    private lateinit var mLinearLayoutManager: LinearLayoutManager
+    private var mIsShowCloseButton = false
+    var namePlayList = ""
+
+    companion object {
+        const val ADD_PLAYLIST = 200
+        val KEY_POSITION = "key_position"
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt]: const ?

File: app/src/main/res/layout/dialog_create_new_playlist.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    android:layout_gravity="center"
+    xmlns:android="http://schemas.android.com/apk/res/android" >
+<TextView
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:text="@string/title_dialog"
+    android:gravity="center"
+    android:textSize="15sp"
+    android:id="@+id/tvDialogCreateNewPlaylist"
+    />
```
Comment (at-trivo) [inline app/src/main/res/layout/dialog_create_new_playlist.xml]: empty tag

File: app/src/main/res/layout/activity_choice_song.xml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            android:layout_height="wrap_content"
+            android:text="@string/choiceAllSOng" />
+
+    </LinearLayout>
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recycleViewPlaylistChoice"
+        android:layout_width="match_parent"
+        android:layout_height="350dp">
+
```
Comment (at-trivo) [inline app/src/main/res/layout/activity_choice_song.xml]: delete line

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import android.widget.EditText
+import internship.asiantech.a2018summerfinal.R
+import internship.asiantech.a2018summerfinal.database.model.Song
+
+class DialogCreatePlaylist() : DialogFragment() {
+    private lateinit var mEdtInputNamePlaylist : EditText
+    lateinit var mListener : EventAddPlaylistListener
+    private val mDataListPlaylists = mutableListOf<Song>()
+
+    val ADD_PLAYLIST_DIALOG_TAG = "add playlist tag"
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt]: ya use this variable for what

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import internship.asiantech.a2018summerfinal.R
+import internship.asiantech.a2018summerfinal.database.model.Song
+
+class DialogCreatePlaylist() : DialogFragment() {
+    private lateinit var mEdtInputNamePlaylist : EditText
+    lateinit var mListener : EventAddPlaylistListener
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt]: private?

File: app/src/main/java/internship/asiantech/a2018summerfinal/ChoiceSongActivity.kt [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    private lateinit var mTvStateChoiceAll: TextView
+    private lateinit var mRecyclerviewPlaylistChoiceSong: RecyclerView
+    private lateinit var mLinearLayoutManager: LinearLayoutManager
+    private var listSongInDevices: MutableList<Song> = mutableListOf()
+    private var mListBoolChoice = mutableListOf<Boolean>()
+    private var listSongInPlaylist = mutableListOf<Song>()
+    private var mIsCheck = false
+
+    companion object {
+        val KEY_POSITION = "key_position"
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/ChoiceSongActivity.kt]: const?

File: app/src/main/java/internship/asiantech/a2018summerfinal/ChoiceSongActivity.kt [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        val RESULT_KEY = "RESULT_KEY"
+    }
+
+    private var mIsCheckAll = false
+    private lateinit var mAdapterChoiceSong: PlaylistChoiceAdapter
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/ChoiceSongActivity.kt]: ya can init adater right here

File: app/src/main/java/internship/asiantech/a2018summerfinal/ChoiceSongActivity.kt [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        mRecyclerviewPlaylistChoiceSong = findViewById(R.id.recycleViewPlaylistChoice)
+        mAdapterChoiceSong = PlaylistChoiceAdapter(mListBoolChoice, listSongInDevices, object : RadioButtonEventChoice {
+            override fun onRadioButtonClickListener(position: Int, ischeck: Boolean) {
+                val music = listSongInDevices.get(position)
+                mListBoolChoice[position] = !mListBoolChoice[position]
+                mAdapterChoiceSong.notifyDataSetChanged()
+            }
+        })
+        mLinearLayoutManager = LinearLayoutManager(this)
+        mRecyclerviewPlaylistChoiceSong.setHasFixedSize(true)
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/ChoiceSongActivity.kt]: try .apply{}

File: app/src/main/java/internship/asiantech/a2018summerfinal/ChoiceSongActivity.kt [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+        } else {
+            for (index in mListBoolChoice.indices) {
+                mListBoolChoice[index] = false
+            }
+        }
+        mAdapterChoiceSong.notifyDataSetChanged()
+    }
+
+    private fun getListChoice(): MutableList<Song> {
+        val mListChoicedSong = mutableListOf<Song>()
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/ChoiceSongActivity.kt]: m is not necessary for local variable

File: app/src/main/java/internship/asiantech/a2018summerfinal/adapter/ListSongInPlaylistAdapter.kt [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    override fun getItemCount(): Int {
+        return listSongInPlaylist.size
+    }
+
+    override fun onBindViewHolder(holder: ListSongInPlayListViewHolder, position: Int) {
+        holder.tvTitleSong?.text = listSongInPlaylist[position].name
+        holder.tvSinger?.text = listSongInPlaylist[position].author
+    }
+
+    inner class ListSongInPlayListViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView) {
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/adapter/ListSongInPlaylistAdapter.kt]: shoundn't use inner class

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt [hunk 13]
```diff
@@ -1,22 +1,130 @@
...
+
+
+    private lateinit var mTvCreatePlaylist: TextView
+    private lateinit var mTvChangePlaylist: TextView
+    private var listPlaylist: MutableList<Playlist> = mutableListOf()
+    private lateinit var mAdapterPlaylist: PlaylistAdapter
+    private lateinit var mRecyclerviewPlaylist: RecyclerView
+    private lateinit var mLinearLayoutManager: LinearLayoutManager
+    private var mIsShowCloseButton = false
+    var namePlayList = ""
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt]: why ya need namePlayList here?

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt [hunk 14]
```diff
@@ -1,22 +1,130 @@
...
+        }
     }
 
-    private fun initViews() {
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == ADD_PLAYLIST && resultCode == RESULT_OK) {
+            val listSongs = data?.getParcelableArrayListExtra<Song>(RESULT_KEY)!!
+            for (song in listSongs) {
+                mAdapterPlaylist.notifyDataSetChanged()
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt]: ya only need to notify data set changed once

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+    private val mDataListPlaylists = mutableListOf<Song>()
+
+    val ADD_PLAYLIST_DIALOG_TAG = "add playlist tag"
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        lateinit var rootView : View
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt]: lateinit ?

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+        rootView = inflater.inflate(R.layout.dialog_create_new_playlist,container,false)
+        return rootView
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+         var builder = AlertDialog.Builder(activity)
+         var rootView : View = View.inflate(activity,R.layout.dialog_create_new_playlist, null)
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt]: did you format file ?

File: app/src/main/java/internship/asiantech/a2018summerfinal/listmusic/ListMusic.kt [hunk 17]
```diff
@@ -10,21 +10,21 @@ import android.support.v4.app.ActivityCompat
...
-    fun getListMusics(): ArrayList<Music> {
+    fun getListMusics(): ArrayList<Song> {
         if (listMusics == null) {
             listMusics = ArrayList()
-            if (ContextCompat.checkSelfPermission(context, Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
-                ActivityCompat.requestPermissions(context as Activity, arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE), REQUEST_PERMISSION)
+            if (ContextCompat.checkSelfPermission(context as Activity, Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+                ActivityCompat.requestPermissions(context , arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE), REQUEST_PERMISSION)
             } else {
                 val selection = MediaStore.Audio.Media.IS_MUSIC + " !=0"
```
Comment (at-trivo) [inline app/src/main/java/internship/asiantech/a2018summerfinal/listmusic/ListMusic.kt]: create a func wound be a good idea

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt (hunk 16 above)
Comment (at-lambui) [reply app/src/main/java/internship/asiantech/a2018summerfinal/fragment/dialog/DialogCreatePlaylist.kt]: I made format file, but it not difference.

File: app/src/main/java/internship/asiantech/a2018summerfinal/listmusic/ListMusic.kt (hunk 17 above)
Comment (at-lambui) [reply app/src/main/java/internship/asiantech/a2018summerfinal/listmusic/ListMusic.kt]: it have completed by Thach.

File: app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt (hunk 13 above)
Comment (at-lambui) [reply app/src/main/java/internship/asiantech/a2018summerfinal/fragment/PlaylistFragment.kt]: I need a global variable to use for another purpose.(in fun addPlaylist())

**Pull request merged**